### PR TITLE
Add EL9-specific Logic for Releasing the Boot-Disk

### LINKED ISSUE #658: [BUG] OS Does Not Fully-Release Boot-Disk
### Expected behavior

Execution of [pivot-root.sh](https://github.com/plus3it/spel/blob/master/spel/scripts/pivot-root.sh) and [umount-sequence](https://github.com/plus3it/spel/blob/master/spel/minimal-linux.pkr.hcl#L918-L936) adequately free up the boot-disk to allow on-the-fly repartitioning.

### Actual behavior

The referenced sequence that worked for EL7 and EL8 distros does not work on EL9

### Steps to reproduce behavior

1. Execute spel-automation
1. Wait for automation to get to the section where it attempts to write a new partiton table and fails
1. Observe an error similar to:
    ~~~
    Error: Partition(s) 4 on /dev/nvme0n1 have been written, but we have been unable to inform the kernel of the change, probably because it/they are in use.  As a result, the old partition(s) will remain in use.  You should reboot now before making further changes.
    ~~~

### Context/Specifications

All RHEL9 and derivative distros

### Suggestions for fix

1. Add logic to explicitly-kill all unnecessary systemd service-units before attempting to null the boot disk.
1. Add logic to explicitly-remove each and every partition on the boot-disk before attempting to null the boot disk.

### Relevant references
- links to "best practices" or other guide(s)
- ...
- ...

### PR DESCRIPTION
**Pull request by ferricoxide**
Fixes #658

### CODE REVIEW

**Comment by ferricoxide** [comment]
@lorengordon: not sure why the Travis-based test is throwing an "unformatted" HCL error? When I run `packer validate`, locally, It comes up green. That said, I'm doing:
```bash
/usr/local/bin/packer validate \
  -only amazon-ebs.minimal-rhel-7-hvm,amazon-ebs.minimal-centos-7-hvm,amazon-ebs.minimal-rhel-8-hvm,amazon-ebs.minimal-centos-8stream-hvm,amazon-ebs.minimal-ol-8-hvm \
  -var "spel_identifier=spel-test" \
  -var "spel_version=2024.01.1-issue_658-01" \
  ./minimal-linux.pkr.hcl
```
Which results in:
```text
The configuration is valid.
```
To validate.

File: spel/minimal-linux.pkr.hcl [hunk 1]
```diff
@@ -923,6 +923,23 @@ build {
       "echo Killing processes locking /oldroot",
       "fuser -vmk /oldroot",
     ]
+    except = [
+      "amazon-ebs.minimal-centos-9stream-hvm",
+      "amazon-ebs.minimal-ol-9-hvm",
+      "amazon-ebs.minimal-rhel-9-hvm",
+    ]
```
Comment (lorengordon) [inline spel/minimal-linux.pkr.hcl:930]: We actually have a similar issue with getting the disk to release on EL7. It's less persistent, more of a race condition, but the presentation is very similar. I think the only reason on EL8 we don't see the problem is due to different handling of the /boot partition.

That saying, I don't think we need to conditionally execute the new script. Just run it on all builders, and remove this provisioner stanza here that is trying to unlock /oldroot.

**Comment by lorengordon** [comment]
> @lorengordon: not sure why the Travis-based test is throwing an "unformatted" HCL error? When I run `packer validate`, locally, It comes up green. That said, I'm doing:

`validate` is not the same as "format". Try `packer fmt -recursive` and commit the change.

**Comment by ferricoxide** [comment]
> > @lorengordon: not sure why the Travis-based test is throwing an "unformatted" HCL error? When I run `packer validate`, locally, It comes up green. That said, I'm doing:
> 
> `validate` is not the same as "format". Try `packer fmt -recursive` and commit the change.

```bash
$ /usr/local/bin/packer fmt -recursive spel
$ echo $?
0
```

**Comment by lorengordon** [comment]
Try `make -f Makefile.tardigrade-ci hcl/format`. That'll actually use terraform under the covers to format any hcl file, same as the way `hcl/lint` works. Terraform fmt must be a little more fascist than packer fmt.

> ```
> $ /usr/local/bin/packer fmt -recursive spel
> $ echo $?
> 0
> 
> $ /usr/local/bin/packer --version
> 1.8.7
> ```

**Review by lorengordon — COMMENTED** [review]
See comment... https://github.com/plus3it/spel/pull/660/files#r1467865910

**Comment by ferricoxide** [comment]
Ah. Ok. Apparently, there was extra whitespace it didn't like. After running the `make`:

~~~bash
$ git diff
diff --git a/spel/minimal-linux.pkr.hcl b/spel/minimal-linux.pkr.hcl
index a511582..60aee4c 100644
--- a/spel/minimal-linux.pkr.hcl
+++ b/spel/minimal-linux.pkr.hcl
@@ -931,7 +931,7 @@ build {
   }

   provisioner "shell" {
-    execute_command   = "{{ .Vars }} sudo -E /bin/sh '{{ .Path }}'"
+    execute_command = "{{ .Vars }} sudo -E /bin/sh '{{ .Path }}'"
     scripts = [
       "${path.root}/scripts/free-root.sh",
     ]
~~~

File: spel/minimal-linux.pkr.hcl (hunk 1 above)
Comment (ferricoxide) [reply spel/minimal-linux.pkr.hcl]: Looks like the logic does nothing to address that and may actually introduce different problems for EL7. Updating the `only`/`exclude` stanzas to ensure that EL7-based builds get the old behavior while everything else gets the new behavior.

**Comment by lorengordon** [comment]
go codebuild go

**Review by lorengordon — APPROVED** [review]

**Pull request merged**